### PR TITLE
adapt ako's fix for avi resource deletion in k8s 1.22

### PR DESCRIPTION
**Pull request by XudongLiuHarold**
Signed-off-by: Xudong Liu <xudongl@vmware.com>

### What this PR does / why we need it

Adapt AKO's fix to use annotations to show the resources deletion status in k8s 1.22.

See this: https://github.com/vmware/load-balancer-and-ingress-services-for-kubernetes/pull/688 for details.

### Which issue(s) this PR fixes

Fixes #

### Describe testing done for PR

- Launched a bolt MR to test full TKG on vsphere lifecycle with this change and it passed.

### Release note

```release-note
This changes to wait for AKO statefulset annotations instead of status for AKO cleanup AVI resources because of AKO k8s 1.22 fixing.
```

### PR Checklist

- [x] Squash the commits into one or a small number of logical commits
      
- [x] Use good commit [messages](https://github.com/vmware-tanzu/tanzu-framework/blob/main/CONTRIBUTING.md)
- [x] Ensure PR contains terms all contributors can understand and links all contributors can access

#### Special notes for your reviewer

/kind bug

### CODE REVIEW

**Review by HanFa — APPROVED** [review]
LGTM thanks @XudongLiuHarold

File: pkg/v1/tkg/clusterclient/resource.go [hunk 1]
```diff
@@ -305,10 +304,8 @@ func VerifyCRSAppliedSuccessfully(obj crtclient.ObjectList) error {
 func VerifyAVIResourceCleanupFinished(obj crtclient.Object) error {
 	switch statefulSet := obj.(type) {
 	case *appsv1.StatefulSet:
-		for _, condition := range statefulSet.Status.Conditions {
-			if condition.Type == constants.AkoCleanupCondition && condition.Status == corev1.ConditionFalse {
-				return nil
-			}
+		if statefulSet.Annotations[constants.AkoCleanUpAnnotationKey] == constants.AkoCleanUpFinishedStatus {
```
Comment (lubronzhan) [inline pkg/v1/tkg/clusterclient/resource.go:307]: First check if annotation is nil?

**Review by lubronzhan — APPROVED** [review]
/lgtm

**Comment by vuil** [comment]
lgtm, thanks!

**Review by vuil — APPROVED** [review]

**Pull request merged**